tests/test_polaraveraging.py

```diff
@@ -0,0 +1,234 @@
+    testing.assert_almost_equal(pa._compute_g_x(0., 100., np.pi/4.), 0.0)
+    testing.assert_almost_equal(pa._compute_g_x(0., 0., 0.3), 0.)
+
+def test_compute_g_t():
+    data = np.array([[0.01, 0.02, 0.01], # g1
+                     [0.01, 0.02, 0.03], # g2
+                     [0.1, 1.2, 3.]]) # phi [rad]
+
+    # test that function works for scalar and vector input
+    # testing.assert(isinstance(float, pa._compute_g_t(*(data[:,0]))))
+    # testing.assert(isinstance(np.array, pa._compute_g_t(*data)))
+    testing.assert_equal(3, len(pa._compute_g_t(*data)))
+    testing.assert_equal(pa._compute_g_t(*(data[:,0])), pa._compute_g_t(*data)[0])
+    
+    # test same length array input
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1,0], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1], data[2,0])
+    
+    # test for input range
+    testing.assert_raises(ValueError, pa._compute_g_t, 0.1, 0.1, -3.15)
+    testing.assert_raises(ValueError, pa._compute_g_t, 0.1, 0.1, 2.*np.pi+0.1)
+    testing.assert_almost_equal(pa._compute_g_t(0.1, 0.1, 0.), pa._compute_g_t(0.1, 0.1, np.pi))
+    
+    # test for reasonable values
+    testing.assert_almost_equal(pa._compute_g_t(100., 0., 0.), -100.0)
+    testing.assert_almost_equal(pa._compute_g_t(0., 100., np.pi/4.), -100.0)
+    testing.assert_almost_equal(pa._compute_g_t(0., 0., 0.3), 0.)
+
+    
+def test_compute_radial_averages():
+    #testing input types
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius="glue", g=10, bins=[np.arange(1.,16.)])
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius=np.arange(1.,10.), g="glue", bins=[np.arange(1.,16.)])  
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius=np.arange(1.,10.), g=np.arange(1.,10.), bins='glue') 
+
+    #want radius and g to have same number of entries
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius=np.arange(1.,10.), g=np.arange(1.,7.), bins=[np.arange(1.,16.)])
+
+    #want binning to encompass entire radial range
+    #testing.assert_raises(UserWarning, pa._compute_radial_averages, radius=np.arange(1.,10.), g=np.arange(1.,10.), bins=[1,6,7])
+    #testing.assert_raises(UserWarning, pa._compute_radial_averages, radius=np.arange(1.,6.), g=np.arange(1.,6.), bins=[5,6,7]) 
+
+    
+
+def test_compute_theta_phi():
+    ra_l, dec_l = 161., 65.
+    ra_s, dec_s = np.array([-355., 355.]), np.array([-85., 85.])
+    rtol=1.e-7
+
+    # Test type on inputs - In retrospect, I don't think these tests really matter much as
```

Comment:
Do we actually want to keep all of the commented out tests?  Does the math break in a predictable way?  It may be helpful to say why they are commented out

Reply:
It looks like these are testing what happens if you pass in the wrong data type. In reality though, if you try to do `'0.5'+.5` your code is going to crash so these tests aren't really needed.